installer/pkg/config-generator/generator.go

```diff
@@ -62,8 +66,91 @@ func New(cluster config.Cluster) ConfigGenerator {
 	}
 }
 
+// maoOperatorConfig contains configuration for mao managed stack
+// TODO(enxebre): move up to "github.com/coreos/tectonic-config
+type maoOperatorConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	TargetNamespace string         `json:"targetNamespace"`
+	APIServiceCA    string         `json:"apiServiceCA"`
+	Provider        string         `json:"provider"`
+	AWS             *awsConfig     `json:"aws"`
+	Libvirt         *libvirtConfig `json:"libvirt"`
+}
+
+type libvirtConfig struct {
+	ClusterName string `json:"clusterName"`
+	URI         string `json:"uri"`
+	NetworkName string `json:"networkName"`
+	Replicas    int    `json:"replicas"`
+}
+
+type awsConfig struct {
+	ClusterName      string `json:"clusterName"`
+	ClusterID        string `json:"clusterID"`
+	Region           string `json:"region"`
+	AvailabilityZone string `json:"availabilityZone"`
+	Image            string `json:"image"`
+	Replicas         int    `json:"replicas"`
+}
+
+func (c *ConfigGenerator) maoConfig(clusterDir string) (*maoOperatorConfig, error) {
+	cfg := maoOperatorConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "machineAPIOperatorConfig",
+		},
+
+		TargetNamespace: maoTargetNamespace,
+	}
+
+	ca, err := ioutil.ReadFile(filepath.Join(clusterDir, aggregatorCACertPath))
```

Comment:
Injecting the CA directly into the config is going to make it difficult to rotate. Please just reference the CA from a well-known secret instead (you'll have to create it).

Reply:
If this is absolutely a blocker, we can find a way to do that. We'll have to read it from somewhere and then inject it into the `APIService` that's registered.

But the reason we need this at all (and a bunch of other annoying stuff) is that this is an aggregated API server. Upstream is planning to move to CRDs. I'd really love to get this merged and start getting actual usage, and then help move away from the aggregated API server to CRDs. At which point this will go away entirely.

Reply:
> We'll have to read it from somewhere and then inject it into the APIService that's registered

Isn't this is exactly what the service-ca-operator does @deads2k ? Will the service-ca-operator be running already?  cc @openshift/sig-security

Reply:
CRDs are on the way https://github.com/kubernetes-sigs/cluster-api/pull/494

Reply:
@crawford i do not think we should block on the aggregator CA being easily rotated.  none of the clusters are forward migratable for the initial milestone.  i am concerned that until we see this working end to end our ability to influence change in the upstream gets smaller and smaller within a valid time window.

Reply:
> @crawford i do not think we should block on the aggregator CA being easily rotated. none of the clusters are forward migratable for the initial milestone. i am concerned that until we see this working end to end our ability to influence change in the upstream gets smaller and smaller within a valid time window.

as @ericavonb noted, we built a controller to do exactly that and an operator to install said controller here https://github.com/openshift/service-serving-cert-signer

Reply:
> Upstream is planning to move to CRDs.

@bison CRDs have severe limitations that make them inappropriate and/or impossible depending the openshift resource you're looking at.  By the time they reach parity, instead of managing one apiserver, you'll be managing at least 4 (validating admission, mutating admission, conversion, subresources) and still be unable to provide useful functionality like `logs`.

They simply don't satisfy the need.

Reply:
I read the comments and not the code.  Isn't this setting https://github.com/openshift/origin/pull/20936/files#diff-6c001ec9fae86f0de6b47aa6b31ce5ddR35 ?  In which the case kube-apiserver itself distributes any changes to every aggregated apiserver?

Reply:
Any change request should be added to the beta requirement doc. Derek, can you make sure we put that in?

Reply:
@eparis yes